docs/src/main/sphinx/sql/pattern-recognition-in-window.rst

```diff
@@ -78,23 +78,24 @@ partition end, and all rows of the partition are available for computations, in
 window structures the pattern matching can neither match rows nor retrieve
 input values outside the frame.
 
-Besides the ``frame_extent``, the two required clauses for pattern matching are
-``PATTERN`` and ``DEFINE``.
+Besides the ``frame_extent``, pattern matching requires the ``PATTERN`` and
+``DEFINE`` clauses.
 
-The ``PATTERN`` clause specifies a row pattern which is a form of a regular
+The ``PATTERN`` clause specifies a row pattern, which is a form of a regular
 expression with some syntactical extensions. The row pattern syntax is similar
 to the :ref:`row pattern syntax in MATCH_RECOGNIZE<row_pattern_syntax>`.
-However, the anchor pattern: ``^`` or ``$`` is not allowed in window.
+However, the anchor patterns ``^`` and ``$`` are not allowed in a window
+specification.
 
-The ``DEFINE`` clause is where row pattern primary variables are defined by
-boolean conditions. It is similar to the
+The ``DEFINE`` clause defines the row pattern primary variables in terms of
+boolean conditions that must be satisfied. It is similar to the
 :ref:`DEFINE clause of MATCH_RECOGNIZE<row_pattern_variable_definitions>`.
 The only difference is that the window syntax does not support the
 ``MATCH_NUMBER`` function.
 
 The ``MEASURES`` clause is syntactically similar to the
-:ref:`MEASURES clause of MATCH_RECOGNIZE<row_pattern_measures>`, the only
-limitation being the usage of the ``MATCH_NUMBER`` function. However, the
+:ref:`MEASURES clause of MATCH_RECOGNIZE<row_pattern_measures>`. The only
+limitation is the usage of the ``MATCH_NUMBER`` function. However, the
```

Comment:
Now that I re-read this, it's not clear what the limitation is. Maybe make it more explicit by saying "The only limitation is that the ``MATCH_NUMBER`` function is not allowed", or something of that sort.